src/awkward/highlevel.py

```diff
@@ -1762,6 +1833,20 @@ def __init__(
         if behavior is not None and not isinstance(behavior, Mapping):
             raise TypeError("behavior must be None or mapping")
 
+        if attrs is not None and not isinstance(attrs, dict):
```

Comment:
Same as @jpivarski's point above :)

Reply:
Sure, I commented under Jims comment already. I resolve this, and we can discuss it in the other discussion thread.